blocks/main_loop.py

```diff
@@ -7,15 +7,33 @@
 from blocks.log import TrainingLog
 from blocks.utils import reraise_as, unpack, change_recursion_limit
 
+logging.basicConfig()
 logger = logging.getLogger(__name__)
 
-error_message = (
-    """An error occurred during the training.
+error_message = """
 
-Attempting to run `on_error` extensions before exiting...""")
+Blocks will attempt to run `on_error` extensions, potentially saving data, \
+before exiting and reraising the error. Note that the usual `after_training` \
+extensions will *not* be run. The original error will be re-raised and also \
+stored in the training log. Press CTRL + C to halt Blocks immediately."""
 
-error_in_error_handling_message = (
-    """While calling `on_error` extension another error occured.""")
+error_in_error_handling_message = """
+
+Blocks will continue running the remaining `on_error` extensions."""
```

Comment:
I think it is not true: after any of the extensions crashes all the rest will simply not be run.